test_simple_graph.py

```diff
@@ -160,3 +160,32 @@ def test_breadth_first_traversal_cyclic(cyclic_graph):
     assert cyclic_graph.breadth_first_traversal(3) == [3, 10, 5, 2, 7]
 
 
+def test_dijkstras_algorithm(full_graph):
+    distance, previous = full_graph.dijkstras_algorithm(3)
+    assert distance[15] == 6
+    assert distance[14] == float('inf')
+    assert previous[15] == 7
+    assert previous[14] is None
+
+
+def test_bellman_ford(full_graph):
+    distance, previous = full_graph.bellman_ford(3)
+    assert distance[15] == 6
+    assert distance[14] == float('inf')
+    assert previous[15] == 7
+    assert previous[14] is None
+
+
+def test_bellman_ford_negative(full_graph):
+    full_graph.add_edge(3, 15, -2)
+    full_graph.add_edge(15, 3, -5)
+    with pytest.raises(ValueError):
+        full_graph.bellman_ford(3)
+
+
+# This produced an infinite cycle, which is what we wanted.
```

Comment:
This is really interesting to me. I wonder if one could have some sort of timer assert? Like if this test lasts more than a set amount of time... It might be infinite.. So pass it.. ? ? That sounds super hacky and pretty smelly so maybe not. You should talk to Cris about this.. And let me know because I want to listen in!